### ISSUE TITLE
Error with distinct for QF_IDL

### ISSUE THREAD

**Issue by MarcRoigVilamala** [SRI-CSL/yices2#94]
I have been using Yices 2.6.0 in order to solve some problems expressed in SMT-LIB 2.

I am using the binary named yices-smt2 and I have observed that, if I set the logic to QF_IDL, an error appears when I use the distinct function for more than two variables. Here is an example:

Input
```
(set-logic QF_IDL)
(declare-const x1 Int)
(declare-const x2 Int)
(declare-const x3 Int)
(assert (distinct x1 x2 x3))
(check-sat)
```

Output
```
(error "formula is not in integer difference logic")
```

I have observed that this error does not appear until I call the function `(check-sat)`. 

The error does not appear if only two variables are used.

I have also tried using the QF_LIA logic and there it seems to work correctly independently of how many variables are being used.

**Comment by MarcRoigVilamala** [comment]
I am quite new to SMT-LIB so I might be getting something wrong, but it was my understanding that the fact that distinct was defined as a pairwise operation in the core (http://smtlib.cs.uiowa.edu/theories-Core.shtml) meant that it applies to all theories.

According to the manual (http://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.6-r2017-07-18.pdf in 3.7 Theory declarations, page 35), if the operation `f` is pairwise that means that `(f t1 ... tn)` is the same as `(and (f t1 t2) ... (f t1 tn) (f t2 ... tn))`, where the last operation is recursive.

I should be able to implement everything I want without using more than two arguments for distinct, but I believe that allowing more than two arguments is the correct syntax.

**Comment by BrunoDutertre** [comment]
Fixed in b8f87fe8f
